crates/typst/src/model/content.rs

```diff
@@ -91,6 +91,18 @@ impl Content {
         Some(self.attrs.iter().filter_map(Attr::child))
     }
 
+    /// Also auto expands sequence of sequences into flat sequence
+    pub fn to_sequence_recursive(&self) -> Option<Box<dyn Iterator<Item = &Self> + '_>> {
+        if !self.is::<SequenceElem>() {
+            return None;
+        }
+        Some(Box::new(
+            self.attrs.iter().filter_map(Attr::child).flat_map(|e| {
+                e.to_sequence_recursive().unwrap_or(Box::new(iter::once(e)))
```

Comment:
especially since this allocates once _per element_ and not per sequence

Reply:
Oh, sorry a lot. That was quite stupid, I don't know what messed with my mind.